modular_zubbers/code/modules/quirks/code/positive_quirks/dominant_aura.dm

```diff
@@ -0,0 +1,76 @@
+/datum/quirk/dominant_aura
+	name = "Dominant Aura"
+	desc = "Your personality is assertive enough to appear as powerful to other people, so much in fact that the weaker kind can't help but throw themselves at your feet on command."
+	icon = "sort-up"
+	medical_record_text = "Patient displays a high level of assertiveness within their personality."
+	value = 1
+	gain_text = span_notice("You feel like making someone your pet.")
+	lose_text = span_notice("You feel less assertive than before")
+	quirk_flags = QUIRK_HUMAN_ONLY | QUIRK_HIDE_FROM_SCAN
+	erp_quirk = TRUE // Disables on ERP config.
+
+/datum/quirk/dominant_aura/add(client/client_source)
+	. = ..()
+	RegisterSignal(quirk_holder, COMSIG_MOB_EXAMINING, PROC_REF(on_sub_examine))
+	RegisterSignal(quirk_holder, COMSIG_MOB_EMOTE, PROC_REF(on_snap))
+
+/datum/quirk/dominant_aura/remove()
+	. = ..()
+	UnregisterSignal(quirk_holder, COMSIG_MOB_EXAMINING)
+	UnregisterSignal(quirk_holder, COMSIG_MOB_EMOTE)
+
+/datum/quirk/dominant_aura/proc/on_sub_examine(atom/source, mob/user, list/examine_list)
+	SIGNAL_HANDLER
+
+	if(!istype(user))
+		return
+	if(user.stat == DEAD)
+		return
+	var/mob/living/carbon/human/sub = user
+	if(!sub.has_quirk(/datum/quirk/well_trained))
+		return
+	examine_list += span_purple("You can sense submissiveness irradiating from them.")
+
+/datum/quirk/dominant_aura/proc/on_snap(atom/source, datum/emote/emote_args)
+	SIGNAL_HANDLER
+
+	INVOKE_ASYNC(src, PROC_REF(handle_snap), source, emote_args)
+
+/datum/quirk/dominant_aura/proc/handle_snap(atom/source, datum/emote/emote_args)
+
+	. = FALSE
+	var/list/emote_list = list("snap", "snap2", "snap3")
+	if(locate(emote_args.key) in emote_list)
+		return
+	if(!TIMER_COOLDOWN_FINISHED(quirk_holder, DOMINANT_COOLDOWN_SNAP))
+		return
+
+	for(var/mob/living/carbon/human/sub in hearers(5, quirk_holder))
+		if(!sub.has_quirk(/datum/quirk/well_trained) || (sub == quirk_holder))
+			continue
+		var/good_x = "pet"
+		switch(sub.gender)
+			if(MALE)
+				good_x = "boy"
+			if(FEMALE)
+				good_x = "girl"
+		switch(emote_args.key)
+			if("snap")
+				sub.dir = get_dir(sub, quirk_holder)
+				sub.emote("me", 1, "faces towards <b>[quirk_holder]</b> at attention!", TRUE)
+				to_chat(sub, span_purple("<b>[quirk_holder]</b>'s snap shoots down your spine and puts you at attention"))
+			if("snap2")
+				sub.dir = get_dir(sub, quirk_holder)
+				sub.Immobilize(0.3 SECONDS)
+				sub.emote("me",1,"hunches down in response to <b>[quirk_holder]'s</b> snapping.", TRUE)
+				to_chat(sub, span_purple("You hunch down and freeze in place in response to <b>[quirk_holder]</b> snapping their fingers"))
+			if("snap3")
+				sub.KnockToFloor(knockdown_amt = 0.1 SECONDS)
+				step(sub, get_dir(sub, quirk_holder))
```

Comment:
I think theres a helper proc for movement that we should be using here, correct me if im wrong

Reply:
I can't think of it at the top of my head. If there's a proc, I'll love to know about it.